apps/system/js/system_banner.js

```diff
@@ -28,29 +34,36 @@
     _clicked: false,
 
     /**
-     * Callback when the user clicks on the system banner button.
+     * Set to true when the banner is currently showing to the user.
      * @memberof SystemBanner.prototype
-     * @type {Function}
+     * @type {Boolean}
      */
-    _clickCallback: null,
+    _showing: false,
 
     /**
      * Generates and returns the banner if it does not exist
      * @memberof SystemBanner.prototype
      * @type {Function}
      */
-    get banner() {
-      if (!this._banner) {
-        this._banner = document.createElement('section');
-        this._banner.className = 'banner generic-dialog';
-        this._banner.setAttribute('role', 'dialog');
-        this._banner.dataset.zIndexLevel = 'system-notification-banner';
-        this._banner.dataset.button = 'false';
-        this._banner.innerHTML = '<p></p><button></button>';
-        document.getElementById('screen').appendChild(this._banner);
+    getBanner: function() {
+      if (this._banner) {
+        return Promise.resolve(this._banner);
       }
 
-      return this._banner;
+      return LazyLoader.load('shared/elements/gaia-component/gaia-component.js')
+        .then(() => {
+          return LazyLoader.load('shared/elements/gaia-toast/gaia-toast.js');
+        })
+        .then(() => {
+          this._banner = document.createElement('gaia-toast');
+          this._banner.timeout = SystemBanner.TIMEOUT;
+          this._banner.className = 'banner';
+          this._banner.dataset.zIndexLevel = 'system-notification-banner';
+          var container = createChild(this._banner, 'div', 'container');
```

Comment:
All the things I am doing here (setting the timeout, z-index level, css class, and child elements) are jobs of the consumer of the web components. By default the web component doesn't need to have any of these things, but I find them helpful with the way SystemBanner uses this web component.